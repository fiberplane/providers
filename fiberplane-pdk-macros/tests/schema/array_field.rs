//! This test makes sure that code does not compile when using a Vec<>
//! in an array field, that uses a non-builtin structure that doesn't
//! have the "schema" class function.

use fiberplane_pdk_macros::QuerySchema;
use serde::Deserialize;

#[derive(Deserialize)]
pub struct RowItem {
    name: String,
    value: String,
}

#[derive(Deserialize, QuerySchema)]
pub struct MissingRowSchema {
    table: Vec<RowItem>,
}

pub fn main() {}
