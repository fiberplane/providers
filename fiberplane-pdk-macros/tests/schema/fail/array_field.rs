//! This test makes sure that code does not compile when using a Vec<T>
//! in an array field, where T is a non-builtin structure that doesn't
//! have the "schema" class function.

use fiberplane_pdk_macros::QuerySchema;
use serde::Deserialize;

#[derive(Deserialize)]
pub struct RowItem {
    name: String,
    value: String,
}

#[derive(Deserialize, QuerySchema)]
pub struct MissingRowSchema {
    table: Vec<RowItem>,
}

pub fn main() {}
