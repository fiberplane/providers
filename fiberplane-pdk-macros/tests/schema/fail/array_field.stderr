error[E0599]: no function or associated item named `schema` found for struct `RowItem` in the current scope
  --> tests/schema/fail/array_field.rs:14:23
   |
9  | pub struct RowItem {
   | ------------------ function or associated item `schema` not found for this struct
...
14 | #[derive(Deserialize, QuerySchema)]
   |                       ^^^^^^^^^^^ function or associated item not found in `RowItem`
   |
   = note: this error originates in the derive macro `QuerySchema` (in Nightly builds, run with -Z macro-backtrace for more info)
